backend/src/ee/services/access-approval-policy/access-approval-policy-service.ts

```diff
@@ -86,13 +87,15 @@ export const accessApprovalPolicyServiceFactory = ({
     if (!groupApprovers && approvals > userApprovers.length + userApproverNames.length)
       throw new BadRequestError({ message: "Approvals cannot be greater than approvers" });
 
-    const { permission } = await permissionService.getProjectPermission(
+    const { permission, ForbidOnInvalidProjectType } = await permissionService.getProjectPermission(
       actor,
       actorId,
       project.id,
       actorAuthMethod,
       actorOrgId
     );
+    ForbidOnInvalidProjectType(ProjectType.SecretManager);
```

Comment:
nit: hard to say from reading it if we reject if type is secrets manager or anything ese. what about `ForbidUnlessProjectType(type)`?

Reply:
Moved to seperate pr